src/org/zaproxy/zap/extension/alert/ExtensionAlert.java

```diff
@@ -376,6 +380,23 @@ private void writeAlertToDB(Alert alert, HistoryReference ref) throws HttpMalfor
 
     }
 
+    private TreePath findAlertInTree(Alert alert) {
+    	Enumeration<?> en = ((((AlertNode) this.getTreeModel().getRoot()).preorderEnumeration()));
+		while (en.hasMoreElements()) {
+			AlertNode node = (AlertNode) en.nextElement();
+			if (node.isRoot()) {
+				continue;
+			}
+			if (alert.compareTo(node.getUserObject())==0) {
+				if (node.getChildCount()==1) {

```

Comment:
Ideally it should continue until it has no child nodes (we should only select leafs).